docs/samples/periodic-tasks.md

```diff
@@ -0,0 +1,248 @@
+
+```sh
+% dfx deploy --argument 10
+[...]
+Deployed canisters.
+URLs:
+   Backend canister via Candid interface:
+      heartbeat: http://127.0.0.1/...
+      timer: http://127.0.0.1/...
+```
+
+- #### Step 5: After 10s, observe similar non-zero counters in both canisters:
+
+```sh
+dfx canister call heartbeat counter
+dfx canister call timer counter
+```
+
+Note, as the canisters deployed one by one, there might be a minor discrepancy in the counters.
+
+Example output:
+
+```sh
+% dfx canister call heartbeat counter               
+(8 : nat32)
+% dfx canister call timer counter    
+(7 : nat32)
+```
+
+- #### Step 6: Compare the amount of cycles used to schedule the periodic task with 10s interval:
+
+```sh
+dfx canister call heartbeat cycles_used
+dfx canister call timer cycles_used
+```
+
+Example output:
+
+```sh
+% dfx canister call heartbeat cycles_used
+(10_183_957 : nat64)
+% dfx canister call timer cycles_used
+(2_112_067 : nat64)
+```
+
+For periodic tasks with 10 sec interval the `heartbeat` canister uses *more* cycles than the `timer` canister.
+
+Not only timers use less cycles, but they also more composable. As there is no global state or methods to export, different libraries with timers could be easily used in the same project.
+
+Also, timers provide isolation between the scheduling logic and the periodic task. If the periodic task fails, all the changes made by this task will be reverted, but the timers library state will be updated, i.e. the failed task will be removed from the list of timers to execute.
+
+For such an isolation of execution and scheduling contexts, internally timers library uses self canister calls:
+
+   ```rust
+   # This is a pseudo-code of a self call:
+   ic_cdk::call(ic_cdk::id(), "periodic_task", ());
+   ```
+
+Despite the [costs](https://internetcomputer.org/docs/current/developer-docs/production/computation-and-storage-costs) associated with such self canister calls, timers library still use less cycles than the heartbeats.
+
+### Example 2: Cycles usage for tasks with 1s interval
+
+
+- #### Step 1: Open a new terminal window in the example root directory:
+
+```sh
+cd examples/rust/periodic_tasks
+```
+
+- #### Step 2: Start a clean local Internet Computer replica and a web server:
+
+```sh
+dfx stop
+dfx start --clean
+```
+
+This terminal will stay blocked, printing log messages, until the `Ctrl+C` is pressed or `dfx stop` command is run.
+
+- #### Step 3: Open another terminal window in the same directory:
+
+```sh
+cd examples/rust/periodic_tasks
+```
+
+Example output:
+
+```sh
+% dfx stop && dfx start --clean
+[...]
+Dashboard: http://localhost:63387/_/dashboard
+```
+
+- #### Step 4:. Compile and deploy `heartbeat` and `timer` canisters, setting the interval for periodic tasks to 1s:
+
+```sh
+dfx deploy --argument 1
+```
+
+The counter will start increasing every second in both canisters.
+
+Example output:
+
+```sh
+% dfx deploy --argument 1
+[...]
+Deployed canisters.
+URLs:
```

Comment:
I wish there was a way to provide an anchor here so that if the terminal output ever changes (formatting, text, etc), we know what to change in the docs...